pkgs/servers/peertube/default.nix

```diff
@@ -0,0 +1,213 @@
+{ ldap ? false, sendmail ? false, light ? null, runCommand, libsass
+, lib, stdenv, rsync, fetchzip, youtube-dl, fetchurl, python, nodejs, nodePackages, yarn2nix-moretea
+, fetchFromGitHub }:
+let
+  nodeHeaders = fetchurl {
+    url = "https://nodejs.org/download/release/v${nodejs.version}/node-v${nodejs.version}-headers.tar.gz";
+    sha256 = "1df3yhlwlvai0m9kvjyknjg11hnw0kj0rnhyzbwvsfjnmr6z8r76";
+  };
+  source = rec {
+    version = "v2.1.1";
+    name = "peertube-${version}";
+    src = fetchFromGitHub {
+      owner = "Chocobozzz";
+      repo = "PeerTube";
+      rev = "76f7b571c04c03ba422bd5790944fe80dbb24067";
+      sha256 = "147gm1j657fkpv2ix1bmkhl7ld5h224q7hgdj9ffj3z14mqgk8hj";
+      fetchSubmodules = true;
+    };
+  };
+  patchedSource = stdenv.mkDerivation (source // rec {
+    phases = [ "unpackPhase" "patchPhase" "installPhase" ];
+    patches = [ ./yarn_fix_http_node.patch ] ++ lib.optionals ldap [ ./ldap.patch ] ++ lib.optionals sendmail [ ./sendmail.patch ];
+    installPhase = ''
+      mkdir $out
+      cp -a . $out/
+    '';
+  });
+  serverPatchedPackage = runCommand "server-package" {} ''
+    mkdir -p $out
+    cp ${patchedSource}/package.json $out/
+    cp ${patchedSource}/yarn.lock $out/
+  '';
+  clientPatchedPackage = runCommand "client-package" {} ''
+    mkdir -p $out
+    cp ${patchedSource}/client/package.json $out/
+    cp ${patchedSource}/client/yarn.lock $out/
+  '';
+
+  yarnModulesConfig = {
+    # all = {
+    #   buildInputs = [ yarn2nix-moretea.yarn2nix.src ];
+    # };
+    bcrypt = {
+      buildInputs = [ nodePackages.node-pre-gyp ];
+      postInstall = let
+        bcrypt_lib = fetchurl {
+          url = "https://github.com/kelektiv/node.bcrypt.js/releases/download/v3.0.7/bcrypt_lib-v3.0.7-node-v64-linux-x64-glibc.tar.gz";
```

Comment:
Again, the package has `platforms = stdenv.lib.platforms.linux;`, but this download link and sha256 are platform specific. In this case, the release only appears to include x86_64 builds.